src/langchain_google_memorystore_redis/doc_saver.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import uuid
+from typing import Optional, Sequence, Set
+
+import redis
+from langchain_core.documents.base import Document
+
+
+class MemorystoreDocumentSaver:
+    """Document Saver for Cloud Memorystore for Redis database."""
+
+    def __init__(
+        self,
+        client: redis.Redis,
+        content_field: str,
+        metadata_fields: Optional[Set[str]] = None,
+        key_prefix: Optional[str] = None,
```

Comment:
should we make this mandatory - otherwise, with the current vector store implementation, I don't think we could create a vector store against documents saved this way.